### PR TITLE
continue extending encrypt module and depricate mcrypt

### PR DESCRIPTION
**Pull request by errotan**

### CODE REVIEW

**Comment by errotan** [comment]
Don't merge yet. More commits to come.

**Comment by errotan** [comment]
Is this enough? What do you think?

**Comment by piotrgolasz** [comment]
Looks good, however old encryption tests were based on a known encryption/decryption table to verify that encryption scheme is correct.

File: modules/encrypt/tests/EncryptMcryptTest.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	/**
+	 * @return void
+	 */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->set_config([
+            'type' => Encrypt_Engine_Mcrypt::TYPE,
+            'cipher' => MCRYPT_RIJNDAEL_128,
```
Comment (piotrgolasz) [inline modules/encrypt/tests/EncryptMcryptTest.php]: I'm not sure if we should use this, as mcrypt is deprecated in php 7. All MCRYPT_ constants are unavailable in php 7.2.
I think we should drop tests for mcrypt and only leave the Mcrypt engine with @deprecated tag.

File: modules/encrypt/tests/EncryptMcryptTest.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            'mode' => MCRYPT_MODE_CBC,
```
Comment (piotrgolasz) [inline modules/encrypt/tests/EncryptMcryptTest.php]: All MCRYPT_ constants are unavailable in PHP7.2
I would suggest dropping tests for Engine Mcrypt leaving only Mcrypt Engine with @deprecated tag.

File: modules/encrypt/tests/EncryptMcryptTest.php (hunk 1 above)
Comment (piotrgolasz) [inline modules/encrypt/tests/EncryptMcryptTest.php]: Drop tests for Engine mcrypt as all MCRYPT_ constants are unavailable in php7.2
Leave only mcrypt engine with @deprecated tag.

**Pull request merged**